Userland/Libraries/LibWebView/SingletonProcessState.cpp

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2024, Andrew Kaster <akaster@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/ByteString.h>
+#include <LibCore/Process.h>
+#include <LibCore/StandardPaths.h>
+#include <LibIPC/ConnectionToServer.h>
+#include <LibWebView/SingletonProcessState.h>
+
+namespace WebView {
+
+static HashMap<int, RefPtr<UIProcessConnectionFromClient>> s_connections;
+
+class UIProcessClient final
+    : public IPC::ConnectionToServer<UIProcessClientEndpoint, UIProcessServerEndpoint> {
+    C_OBJECT(UIProcessClient);
+
+private:
+    UIProcessClient(NonnullOwnPtr<Core::LocalSocket>);
+};
+
+ErrorOr<SingletonProcessState::ProcessDisposition> SingletonProcessState::connect(Vector<StringView> const& raw_urls, bool new_window)
+{
+    static constexpr auto process_name = "Ladybird"sv;
+
+    auto runtime_directory = TRY(Core::StandardPaths::runtime_directory());
+    auto socket_path = ByteString::formatted("{}/{}.socket", runtime_directory, process_name);
+    auto pid_path = ByteString::formatted("{}/{}.pid", runtime_directory, process_name);
+
+    if (auto pid = TRY(Core::get_process_pid(process_name, pid_path)); pid.has_value()) {
+        Vector<ByteString> urls;
+        urls.ensure_capacity(raw_urls.size());
+        for (auto const& url : raw_urls)
+            urls.unchecked_append(url.to_byte_string());
```

Comment:
Instead of having to do this, can we just make the function accept `Vector<ByteString>`? And then as each chrome is ported to this, we can change its command line storage to match:
https://github.com/SerenityOS/serenity/blob/f16f89eb32714195b977b3458052f5652e2e2ddf/Ladybird/Qt/main.cpp#L79